clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -52,13 +53,32 @@
 
     private final ConfigDef definition;
 
+    private static final String CONFIG_PROVIDERS_CONFIG = "config.providers";
+
+    /**
+     * Construct a configuration with a ConfigDef, the configuration properties, optional {ConfigProviders}.
+     * @param definition the definition of the configurations
+     * @param originals the name-value pairs of the configuration
+     * @param configProviders the map of properties of config providers which will be instantiated by the constructor to resolve any variables in {@code originals}
+     * @param doLog whether the configurations should be logged
+     * @param enableAutoResolution to enable/disable automatic resolution of indirect variable
+     */
     @SuppressWarnings("unchecked")
-    public AbstractConfig(ConfigDef definition, Map<?, ?> originals, boolean doLog) {
```

Comment:
Looks like this constructor was removed? We can't remove public constructors from a class in the public API.